roles/node/tasks/config-chains.yml

```diff
@@ -17,3 +17,15 @@
     group: "{{ avalanchego_group }}"
   loop: "{{ avalanchego_chains_configs | dict2items }}"
   notify: Restart avalanchego
+
+- name: Generate chains upgrade.json
+  copy:
+    src: "{{ files_path }}/{{ avalanchego_network_id }}/vms/{{ item }}/upgrades.json"
+    dest: "{{ avalanchego_chains_conf_dir }}/{{ item }}/upgrade.json"
+    owner: "{{ avalanchego_user }}"
+    group: "{{ avalanchego_group }}"
+  loop: "{{ avalanchego_chains_configs.keys() }}"
+  vars:
+    _file: "{{ files_path }}/{{ avalanchego_network_id }}/vms/{{ item }}/upgrades.json"
+  when: _file is exists
+  notify: Restart avalanchego
```

Comment:
What would you think of using a similar approach to chains/subnets config rather than uploading files:
```yaml
avalanchego_chains_updates:
  XXX:
    precompileUpgrades:
      - feeManagerConfig:
          blockTimestamp: 1668950000
          adminAddresses:
            - 0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC
      - txAllowListConfig:
          blockTimestamp: 1668960000
          adminAddresses:
            - 0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC
      - feeManagerConfig:
          blockTimestamp: 1668970000
          disable: true
```
IMO this would greatly enhance maintainability (when using GitOps best practices).

You can look at [roles/node/tasks/config-chains.yml](https://github.com/AshAvalanche/ansible-avalanche-collection/blob/main/roles/node/tasks/config-chains.yml) for how to generate a JSON file out of an Ansible variable

Reply:
I thought of this one but I was a bit lazy.  I agree this is better.